### ISSUE TITLE
Verify first boot of appliance with no nic will correctly setup or delay database setup until next boot

### ISSUE THREAD

**Issue by jrafanie** [ManageIQ/manageiq-appliance-build#55]
If you forget or your hypervisor doesn't give you a NIC (I'm looking at you RHEV), will the first-boot service to setup the database run correctly or prevent disabling itself until it's successful on boot?

https://github.com/ManageIQ/manageiq-appliance-build/blob/dd81e593a7ea12e0a1925f1dfcd79c52a32204d0/kickstarts/partials/post/db_init.ks.erb#L21

cc @carbonin @abellotti 

I don't remember who added it but if you would take the liberty of changing the name from `appliance-initialize.service` to `setup-database-on-first-boot.service` or something I would :heart: it.

**Comment by jrafanie** [comment]
This is based on an issue that @brandondunne ran into, maybe he can help illustrate or clarify the use case.

**Comment by abellotti** [comment]
As to the name, that'll be me :)  but I'd keep it generic just in case we need to do something other than database setup on first boot, we don't want to end up with multiple first-boot services. Maybe something like initialize-appliance-on-first-boot.service ?

**Comment by carbonin** [comment]
I would imagine because it will always use the internal disk for database configuration that it wouldn't care about network connectivity.  What would happen if it did fail for some other reason is a good question though.  Would it disable the service even if the `ExecStart=` script fails?

**Comment by brandondunne** [comment]
I think it may have started the script, then I shut it down while it was still initializing.  After that it was in a partially initialized state which it couldn't recover from.

**Comment by carbonin** [comment]
Right, probably due to https://github.com/ManageIQ/manageiq-appliance-build/blob/dd81e593a7ea12e0a1925f1dfcd79c52a32204d0/kickstarts/partials/post/db_init.ks.erb#L4

**Comment by abellotti** [comment]
Right, I needed to make 100% sure, even if they manually re-enable the
service, that we don't blow away a database.

On Wed, Sep 16, 2015 at 10:10 AM, Nick Carboni <notifications@github.com>
wrote:

> Right, probably due to
> https://github.com/ManageIQ/manageiq-appliance-build/blob/dd81e593a7ea12e0a1925f1dfcd79c52a32204d0/kickstarts/partials/post/db_init.ks.erb#L4
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq-appliance-build/issues/55#issuecomment-140752752>
> .
>



-- 
*Alberto Bellotti*
abellott@redhat.com

**Comment by JPrause** [comment]
@abellotti @jrafanie does this issue as described still exist, or is this now resolved. (checking on issues with no recent updates)

**Comment by simaishi** [comment]
Tried this with the latest appliance. As @carbonin mentioned, database setup finished successfully even without network.

I also tried shutting down appliance during database setup. The shutdown happened after initializing postgresql but before creating region. Upon booting up again, I was able to use reset option to get working database and start evmserverd.

If we find a specific case where we can't recover, we can open a new issue. Closing.
